diff --git a/node_modules/@stencil/core/compiler/stencil.js b/node_modules/@stencil/core/compiler/stencil.js
index b459aba..463e932 100644
--- a/node_modules/@stencil/core/compiler/stencil.js
+++ b/node_modules/@stencil/core/compiler/stencil.js
@@ -57211,7 +57211,7 @@ const getSystemLoader = async (config, compilerCtx, corePath, includePolyfills)
 
   var resourcesUrl = scriptElm ? scriptElm.getAttribute('data-resources-url') || scriptElm.src : '';
   var start = function() {
-    var url = new URL('${corePath}', new URL(resourcesUrl, window.location.origin));
+    var url = new URL('${corePath}', new URL(resourcesUrl));
     System.import(url.href);
   };
 
