diff --git a/node_modules/@stencil/core/compiler/stencil.js b/node_modules/@stencil/core/compiler/stencil.js
index 8e21dfb..63c663e 100644
--- a/node_modules/@stencil/core/compiler/stencil.js
+++ b/node_modules/@stencil/core/compiler/stencil.js
@@ -56322,7 +56322,7 @@ const getSystemLoader = async (config, compilerCtx, corePath, includePolyfills)
 
   var resourcesUrl = scriptElm ? scriptElm.getAttribute('data-resources-url') || scriptElm.src : '';
   var start = function() {
-    var url = new URL('${corePath}', new URL(resourcesUrl, window.location.origin));
+    var url = new URL('${corePath}', new URL(resourcesUrl));
     System.import(url.href);
   };
 
diff --git a/node_modules/@stencil/core/testing/index.js b/node_modules/@stencil/core/testing/index.js
index 2ece03f..efb3e08 100644
--- a/node_modules/@stencil/core/testing/index.js
+++ b/node_modules/@stencil/core/testing/index.js
@@ -750,7 +750,7 @@ async function addE2EListener(e, t, r, s) {
   eventName: r,
   callback: s
  });
- const i = t.executionContext();
+ const i = t.realm;
  await i.evaluate(((e, t, r) => {
   e.addEventListener(r, (e => {
    window.stencilOnEvent(t, window.stencilSerializeEvent(e));
@@ -844,14 +844,14 @@ async function findAll(e, t, r) {
  const s = [], {lightSelector: n, shadowSelector: i} = getSelector(r), o = await t.$$(n);
  if (0 === o.length) return s;
  if (i) for (let t = 0; t < o.length; t++) {
-  const r = o[t].executionContext(), n = await r.evaluateHandle(((e, t) => {
+  const r = o[t].realm, n = await r.evaluateHandle(((e, t) => {
    if (!e.shadowRoot) throw new Error(`shadow root does not exist for element: ${e.tagName.toLowerCase()}`);
    return e.shadowRoot.querySelectorAll(t);
   }), o[t], i);
   await o[t].dispose();
   const a = await n.getProperties();
   await n.dispose();
-  for (const t of a.values()) {
+  for (const t of Array.from(a.values())) {
    const r = t.asElement();
    if (r) {
     const t = new E2EElement(e, r);
@@ -3134,7 +3134,7 @@ class E2EElement extends index_cjs.MockHTMLElement {
   this._validate();
   let e = !1;
   try {
-   const t = this._elmHandle.executionContext();
+   const t = this._elmHandle.realm;
    e = await t.evaluate((e => new Promise((t => {
     window.requestAnimationFrame((() => {
      if (e.isConnected) {
@@ -3183,7 +3183,7 @@ class E2EElement extends index_cjs.MockHTMLElement {
  }
  async getProperty(e) {
   this._validate();
-  const t = this._elmHandle.executionContext();
+  const t = this._elmHandle.realm;
   return await t.evaluate(((e, t) => e[t]), this._elmHandle, e);
  }
  setProperty(e, t) {
@@ -3340,7 +3340,7 @@ class E2EElement extends index_cjs.MockHTMLElement {
  }
  async e2eRunActions() {
   if (0 === this._queuedActions.length) return;
-  const e = this._elmHandle.executionContext(), t = await e.evaluate(((e, t) => e.componentOnReady().then((() => {
+  const e = this._elmHandle.realm, t = await e.evaluate(((e, t) => e.componentOnReady().then((() => {
    let r = null;
    return t.forEach((t => {
     if (t.methodName) r = e[t.methodName].apply(e, t.methodArgs); else if (t.setPropertyName) e[t.setPropertyName] = t.setPropertyValue; else if (t.setAttributeName) e.setAttribute(t.setAttributeName, t.setAttributeValue); else if (t.removeAttribute) e.removeAttribute(t.removeAttribute); else if (t.toggleAttributeName) "boolean" == typeof t.toggleAttributeForce ? e.toggleAttribute(t.toggleAttributeName, t.toggleAttributeForce) : e.toggleAttribute(t.toggleAttributeName); else if (t.classAdd) e.classList.add(t.classAdd); else if (t.classRemove) e.classList.remove(t.classRemove); else if (t.classToggle) e.classList.toggle(t.classToggle); else if (t.eventName) {
@@ -3355,7 +3355,7 @@ class E2EElement extends index_cjs.MockHTMLElement {
   return this._queuedActions.length = 0, t;
  }
  async e2eSync() {
-  const e = this._elmHandle.executionContext(), {outerHTML: t, shadowRootHTML: r} = await e.evaluate((e => ({
+  const e = this._elmHandle.realm, {outerHTML: t, shadowRootHTML: r} = await e.evaluate((e => ({
    outerHTML: e.outerHTML,
    shadowRootHTML: e.shadowRoot ? e.shadowRoot.innerHTML : null
   })), this._elmHandle);
